src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -251,9 +252,17 @@ private function guessHandledClasses(\ReflectionClass $handlerClass, string $ser
     private function registerReceivers(ContainerBuilder $container, array $busIds)
     {
         $receiverMapping = [];
+        $failureTransportsMap = [];
+
+        $globalFailureTransportId = 'messenger.failure_transports.default';
+        if ($container->hasAlias($globalFailureTransportId)) {
+            $globalFailureTransport = (string) $container->getAlias($globalFailureTransportId);
+            $failureTransportsMap[$globalFailureTransport] = new Reference('messenger.transport.'.$globalFailureTransport);
```

Comment:
This looks odd to me. If I understand it correctly, `$globalFailureTransport` will equal `messenger.failure_transports.default`, right? Why do we want that?

I understand that we're passing the "global failure receiver/transport" name to the 1st argument of the commands. And I understand that the service locator we're creating here will need to contain a receiver with that name. So, if my global failure receiver is called "failure_main", then `failure_main` will be passed as the first argument to the commands and we *also* want this `$failureTransportsMap` to have a `failure_main` key inside it, right?

If so, I think this isn't quite right yet. It's a bit odd, but could we instead actually fetch one of the command definitions - e.g. `console.command.messenger_failed_messages_retry` and *read* the `0` argument and use *that* as the key? So if the 0 argument is set to `failure_main`, then we would read that and use `failure_main` here. Also, for the `$value, we can just use the `messenger.failure_transports.default` alias. So something like:

```php
$commandDefinition = $container->getDefinition('console.command.messenger_failed_messages_retry');
$failureTransportsMap[$commandDefinition->getArgument(0)] = new Reference('messenger.failure_transports.default');
```